examples/java/src/main/java/org/apache/beam/examples/complete/datatokenization/transforms/io/TokenizationFileSystemIO.java

```diff
@@ -141,12 +149,26 @@ public TokenizationFileSystemIO(DataTokenizationOptions options) {
       case CSV:
         return readCsv(pipeline, schema)
             .apply(new JsonToBeamRow(options.getNonTokenizedDeadLetterPath(), schema));
+      case AVRO:
+        return readAvro(pipeline, schema);
       default:
         throw new IllegalStateException(
             "No valid format for input data is provided. Please, choose JSON or CSV.");
     }
   }
 
+  private PCollection<Row> readAvro(Pipeline pipeline, SchemasUtils schema) {
+    org.apache.avro.Schema avroSchema = AvroUtils.toAvroSchema(schema.getBeamSchema());
+    PCollection<GenericRecord> genericRecords = pipeline.apply(
+        "ReadAvroFiles",
+        AvroIO.readGenericRecords(avroSchema).from(options.getInputFilePattern()));
+    return genericRecords
+        .apply(
+            "GenericRecordToRow", MapElements.into(TypeDescriptor.of(Row.class))
+                .via(AvroUtils.getGenericRecordToRowFunction(schema.getBeamSchema())))
+        .setCoder(RowCoder.of(schema.getBeamSchema()));
+  }
```

Comment:
Let's merge these 2 parts into 1 the same way it is done in `writeAvro` function.

Reply:
done